flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectQueryITCase.java

```diff
@@ -1062,6 +1062,63 @@ public void testSumAggWithGroupKey() throws Exception {
         tableEnv.executeSql("drop table test_sum_group");
     }
 
+    @Test
+    public void testAvgAggFunctionPlan() {
+        // test explain
+        String actualPlan = explainSql("select x, avg(y) from foo group by x");
+        assertThat(actualPlan).isEqualTo(readFromResource("/explain/testAvgAggFunctionPlan.out"));
+    }
+
+    @Test
+    public void testAvgAggFunction() throws Exception {
+        tableEnv.executeSql(
+                "create table test_avg(a int, x string, y string, z int, f bigint, d decimal(20, 5), e double)");
+        tableEnv.executeSql(
+                        "insert into test_avg values (1, NULL, '2', 1, 2, 2.22, 2.3), "
+                                + "(1, NULL, 'b', 2, NULL, 3.33, 3.4), "
+                                + "(2, NULL, '4', 1, 2, 4.55, 4.5), "
+                                + "(2, NULL, NULL, 4, 3, 5.66, 5.2)")
+                .await();
+
+        // test avg all element is null
+        List<Row> result =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select avg(x) from test_avg").collect());
+        assertThat(result.toString()).isEqualTo("[+I[null]]");
+
+        // test avg that some string elements can't convert to double
+        List<Row> result2 =
+                CollectionUtil.iteratorToList(
+                        tableEnv.executeSql("select avg(y) from test_avg").collect());
+        assertThat(result2.toString()).isEqualTo("[+I[3.0]]");
```

Comment:
should be `[+I[2.0]]`

Reply:
As the following explains, it should be `[+I[3.0]]`, the null element doesn't count.

Reply:
Sorry for that. Seems there's some issue in my EMR Hive cluster. It may be not community compatiable.
Test with Hive2 & Hive 3 with Hive's own implementation for avg, the result should be `[+I[2.0]]`.